src/app/dashboard/constraints-overlay/constraints-overlay.component.ts

```diff
@@ -1,102 +1,46 @@
-  noFeasibleSolutionHintShown = false;
-  protected noFeasibleSolutionHintTimeoutHandle = null;
-  globalConstraints: Constraint[];
-  teamConstraints: Record<string, Constraint[]>;
+export class ConstraintsOverlayComponent implements OverlayComponent, OnInit {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
 
   constructor(
-    private constraintService: ConstraintService,
-    private teamGenerationService: TeamGenerationService,
-    public teamService: TeamService,
-    private overlayService: OverlayService
+    private constraintsService: ConstraintsService,
+    private mandatoryConstraintsService: MandatoryConstraintsService,
+    private costFunctionsService: CostFunctionsService,
+    private matchingService: MatchingService,
+    private studentsService: StudentsService,
+    private allocationsService: AllocationsService
   ) {}
 
-  ngOnInit() {
-    this.constraintService.fetchConstraints().then(constraints => {
-      this.constraints = constraints;
-      this.globalConstraints = this.getGlobalConstraints();
-      this.teamConstraints = this.groupByTeam(this.constraints);
-    });
-  }
-
-  ngOnDestroy(): void {
-    this.constraintService.saveConstraints(this.constraints);
-  }
-
-  public getGlobalConstraints(): Array<Constraint> {
-    if (!this.constraints) return [];
-    return this.constraints.filter(constraint => {
-      return constraint.getTeamName() === null;
-    });
-  }
-  public applyConstraints() {
-    this.constraintService.saveConstraints(this.constraints);
-    this.teamService.resetUnpinnedPersons();
-
-    this.teamGenerationService.generate(this.teamService.persons, this.teamService.teams).then(feasible => {
-      if (!feasible) {
-        clearTimeout(this.noFeasibleSolutionHintTimeoutHandle);
-        this.noFeasibleSolutionHintShown = true;
+  constraints: ConstraintWrapper[];
 
-        this.noFeasibleSolutionHintTimeoutHandle = setTimeout(
-          () => (this.noFeasibleSolutionHintShown = false),
-          ConstraintsOverlayComponent.noFeasibleSolutionHintTimeout
-        );
-        return;
-      }
-
-      this.logConstraints();
-      this.teamService.updateDerivedProperties();
-      this.overlayService.closeOverlay();
-      this.teamService.saveToLocalBrowserStorage();
-    });
+  ngOnInit() {
+    this.constraints = this.constraintsService.getConstraints();
   }
 
-  private logConstraints() {
-    const loggedMessage: string[] = [];
-
-    loggedMessage.push(new Date().toUTCString() + ' - Persons distributed with constraints:');
-    loggedMessage.push(
-      ...this.constraints.filter(constraint => constraint.isEnabled).map(constraint => constraint.toString())
-    );
-
-    ConstraintLoggingService.pushMessage(loggedMessage.join('\r\n'));
+  async distributeTeams() {
+    const constraints = this.constraints.flatMap(constraint => constraint.constraint);
+    constraints.push(...this.mandatoryConstraintsService.constraints);
```

Comment:
`constraints.push(...this.mandatoryConstraintsService.constraints, ...this.costFunctionsService.constraints);`